scrapers/bank_of_mauritius.py

```diff
@@ -0,0 +1,81 @@
+from typing import List
+
+import requests
+from playwright.sync_api import sync_playwright
+from utils import (
+    download_directory,
+    get_browser,
+    get_file_path,
+    get_md5hash,
+    get_standardized_filename,
+    parse_arguments,
+)
+
+BASE_URL = "https://www.bom.mu"
+CIRCULAR_PATH = "/financial-stability/supervision/guideline"
+
+
+def get_pdf_links(base_url: str, circular_path: str, count: int = 1) -> List[str]:
+    """Scraper function for circulars published by Bank of Mauritius
+
+    Args:
+        base_url (str): Home page of bank
+        circular_path (str): URL path containing circular publications
+        count (int, optional): Number of most recent circulars desired. Defaults to 1.
+
+    Returns:
+        List[str]: The URL path of the PDF
+    """
+    with sync_playwright() as pw:
+        browser = get_browser(pw)
+        page = browser.new_page()
+        page.goto(f"{base_url}{circular_path}")
+
+        # Since the PDFs are linked in each page individually,
+        # need to first page the page link
+        page_links = page.locator("div.col-sm-8 > a").all()[0:count]
+        links = [page_link.get_attribute("href") for page_link in page_links]
+
+        pdf_links = []
+        for link in links:
+            page.goto(f"{base_url}{link}")
+            pdf_icons = page.locator('img[src="/sites/default/files/pdf-icon.png"]').all()
+            pdf_links += [
+                icon.locator("..").get_attribute("href")
+                for icon in pdf_icons
+                if icon.locator("..").get_attribute("href").endswith(".pdf")
+            ]
```

Comment:
I am generally cool with one-liners, and I notice that devs who write more Java have much higher comfort around complex one-liners.

But here we are doing `icon.locator("..").get_attribute("href")` twice - that is a complexity that is arguably harder to maintain.

```suggestion
      for icon in pdf_icons:
          href = icon.locator("..").get_attribute("href")
          # Unsure if the walrus operator would work here
          # If yes, it would be pretty cool lol
          if href.endswith(".pdf"):
              pdf_links.append(href)
```

Reply:
Thanks for the suggestion!  I had defaulted to a list comprehension for speed, but I agree that the clarity outweighs the minor speed gain here.